### ISSUE TITLE
Cannot create a project that uses Quarto

### ISSUE THREAD

**Issue by StuieT85** [rstudio/rstudio#11763]
### System details

    RStudio Edition : desktop
    RStudio Version : 2022.07.01
    OS Version      :  Windows 10 21H2
    R Version       :  4.2.1

### Steps to reproduce the problem
Create a new project of any type that starts with Quarto

### Describe the problem in detail
When I attempt to create the project a 'Creating project...' box pops up with "'C:/Program' is not recognized as an internal or external command, operable program or batch file."

### Describe the behavior you expected
A project to be created

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by StuieT85** [comment]
I have just tried all of the other options and they worked without issue. I don't know if it helps but the directory for the project is created and it contains an Rproj file. When I try to open the Rproj file manually I get the message 'The system cannot find the file specified'. The project seems to open. All subsequnt openings of the project are issue three. 

There are no other files in the project though. There are no qmd files just an Rproj  file

**Comment by IanWorthington** [comment]
I've just updated to the latest Rstudio and am having this as well.

Problem seems slightly reminiscent of https://github.com/rstudio/DT/issues/98

I have Rstudio at C:\Program Files\RStudio\bin\rstudio.exe

and R at C:\Program Files\R\R-4.1.2

so either of these might be possible reasons.

**Comment by IanWorthington** [comment]
I've just updated to the latest Rstudio and am having this as well.

Problem seems slightly reminiscent of https://github.com/rstudio/DT/issues/98

I have Rstudio at C:\Program Files\RStudio\bin\rstudio.exe

and R at C:\Program Files\R\R-4.1.2

so either of these might be possible reasons

**Comment by IanWorthington** [comment]
TL;DR: Seems to be a regression in the Windows Rstudio preventing this version from working properly when installed in the standard Windows "Program Files" directory.

Temporary workaround: resinstall to a path without spaces in the directory names.

**Comment by cderv** [comment]
To add to this, report in https://github.com/quarto-dev/quarto-cli/issues/2208 shares the same conclusion as in https://github.com/rstudio/rstudio/issues/11763#issuecomment-1229498963 

> Temporary workaround: resinstall to a path without spaces in the directory names.

Installing the RStudio IDE into a directory without space in name seems to fix the issue.

**Comment by kevinushey** [comment]
The problem here is likely with how we're quoting arguments in CreateProcess here:

https://github.com/rstudio/rstudio/blob/main/src/cpp/core/system/Win32ChildProcess.cpp#L530-L544

Because we're running `quarto.cmd` rather than `quarto.exe` directly, we need to perform some extra escaping to ensure the command line "survives" when it passes through `cmd.exe`.

Here's how the command line appears:

> "C:\Program Files\RStudio\resources\app\bin\quarto\bin\quarto.cmd" preview "Has Spaces.qmd" --to html --no-watch-inputs --no-browse

And indeed, passing that along as an invocation through `cmd.exe` fails:

```
C:\Users\kevin>cmd.exe /C "C:\Program Files\RStudio\resources\app\bin\quarto\bin\quarto.cmd" preview "Has Spaces.qmd" --to html --no-watch-inputs --no-browse
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
```

Unfortunately, quoting the whole command (as is normally suggested) also fails:

````
C:\Users\kevin>cmd.exe /C ""C:\Program Files\RStudio\resources\app\bin\quarto\bin\quarto.cmd" preview "Has Spaces.qmd" --to html --no-watch-inputs --no-browse"
error: Found argument 'Files\RStudio\resources\app\bin\quarto\bin\\tools\deno.exe' which wasn't expected, or isn't valid in this context
```

The daily builds of RStudio try to side-step this issue by using a short path name to `quarto.cmd`.

**Comment by kevinushey** [comment]
I believe so (since most of the escaping issues we're running into here are a result of our command being run through `cmd.exe` which has its own esoteric set of rules for quotation).

**Comment by IanWorthington** [comment]
Oddly enough my work machine did not suffer from this annoyance initially, 2 days ago, but does today. Bloody Windows.  

But they're giving me a new laptop tomorrow though with something better than an i3 with 8Gb so I get the joy of reinstalling everything anyway.  Is there a new version of RStudio with this fixed I can pick up?

**Comment by mhweber** [comment]
Same problem rendering quarto documents- removing RStudio installed in C:\Program Files\RStudio and installing in C:\RStudio fixed the problem and quarto renders now.

**Comment by QuinnAsena** [comment]
I had this problem in VS code. Solved itself when I checked for a Quarto CLI update and realised I didn't have [Quarto CLI](https://quarto.org/docs/download/release.html) installed on this computer. Quarto works with the VS code extension without the CLI installed? Could have sworn I had both...

**Comment by cderv** [comment]
@ronblum @kevinushey I believe `quarto.exe` is now used
https://github.com/rstudio/rstudio/blob/57fdf5da2ce3d58f28cbb461ae5c5647c448cae8/src/cpp/session/modules/quarto/SessionQuarto.cpp#L231-L236

So probably this issue is no more happening.

**Comment by ronblum** [comment]
Adding to Desert Sunflower to check whether or not this is now all set.

**Comment by ronblum** [comment]
Verified in RStudio Desktop 2023.9.0-daily+364 on Windows 11. Note that Windows takes care of spaces in the path to `quarto.exe`
```
> quarto::quarto_path()
[1] "C:\\PROGRA~1\\RStudio\\RESOUR~1\\app\\bin\\quarto\\bin\\quarto.exe"
```
